internal/pkg/infrastructure/database/database.go

```diff
@@ -10,9 +10,14 @@ import (
 )
 
 //go:generate moq -rm -out storage_mock.go . Storage
+//go:generate moq -rm -out func_storage_mock.go . FuncStorage
```

Comment:
Does this work for you? I thought the go:generate had to be just before the interface declaration.

Reply:
Sure do for me